index.js

```diff
@@ -983,131 +1045,204 @@
     return x => this (f (x));
   }
 
+  const staticImplementations = {
+    String: {
+      empty: String$empty,
+    },
+    Array: {
+      empty: Array$empty,
+      of: Array$of,
+      chainRec: Array$chainRec,
+      zero: Array$zero,
+    },
+    Object: {
+      empty: Object$empty,
+      zero: Object$zero,
+    },
+    Function: {
+      id: Function$id,
+      of: Function$of,
+      chainRec: Function$chainRec,
+    },
+  };
+
   const staticMethod = (name, typeRep) => {
-    switch (typeRep.name + '.' + name) {
-      case 'String.fantasy-land/empty':
-        return String$empty;
-      case 'Array.fantasy-land/empty':
-        return Array$empty;
-      case 'Array.fantasy-land/of':
-        return Array$of;
-      case 'Array.fantasy-land/chainRec':
-        return Array$chainRec;
-      case 'Array.fantasy-land/zero':
-        return Array$zero;
-      case 'Object.fantasy-land/empty':
-        return Object$empty;
-      case 'Object.fantasy-land/zero':
-        return Object$zero;
-      case 'Function.fantasy-land/id':
-        return Function$id;
-      case 'Function.fantasy-land/of':
-        return Function$of;
-      case 'Function.fantasy-land/chainRec':
-        return Function$chainRec;
-      default:
-        return null;
+    switch (typeRep) {
+      case String: return staticImplementations.String[name];
+      case Array: return staticImplementations.Array[name];
+      case Object: return staticImplementations.Object[name];
+      case Function: return staticImplementations.Function[name];
+    }
```

Comment:
Are you in favour of aligning these `return` statements?

```javascript
switch (typeRep) {
  case String:      return staticImplementations.String[name];
  case Array:       return staticImplementations.Array[name];
  case Object:      return staticImplementations.Object[name];
  case Function:    return staticImplementations.Function[name];
}

const prefixedName = 'fantasy-land/' + name;
if (typeof typeRep[prefixedName] === 'function') {
  return typeRep[prefixedName];
}

switch (typeRep.name) {
  case 'String':    return staticImplementations.String[name];
  case 'Array':     return staticImplementations.Array[name];
  case 'Object':    return staticImplementations.Object[name];
  case 'Function':  return staticImplementations.Function[name];
}
```

Reply:
My preference would be not to align them. But I don't feel all that strong about it.